### ISSUE TITLE
Shutters not working with USB300 (ESP3)

### ISSUE THREAD

**Issue by grimmpp** [grimmpp/home-assistant-eltako#96]
Sent by [tden159](https://community.home-assistant.io/u/tden159) in [HA Community](https://community.home-assistant.io/t/eltako-baureihe-14-rs485-enocean-debugging/49712/355?u=philipp14).

after the release of v1.4. i tried out your solution with an USB300 as my gateway. I was able to integrate all my lights without any problems. However with my cover/shutter integration i have a problem which i cant solve on my own.

Logs:
```
2024-04-14 08:06:54.961 ERROR (Thread-5) [enocean.communicators.SerialCommunicator] Received ESP3 response with with return code WRONG_PARAM (3) - 0x02 ['0x3'] [] OrderedDict() 
2024-04-14 08:06:56.771 ERROR (Thread-5) [enocean.communicators.SerialCommunicator] Received ESP3 response with with return code WRONG_PARAM (3) - 0x02 ['0x3'] [] OrderedDict() 
2024-04-14 08:06:58.484 ERROR (Thread-5) [enocean.communicators.SerialCommunicator] Received ESP3 response with with return code WRONG_PARAM (3) - 0x02 ['0x3'] [] OrderedDict() 
```

My setup:
HA OS → USB300 → FAM14 → FSB14
Eltako v1.4.2

My config (abstract):

```
eltako:
  gateway:
  - id: 1
    base_id: FF-88-03-00
    device_type: enocean-usb300 
    devices: 

        light:
        - id: FF-8F-AB-01
          name: F4SR14_LED FF-8F-AB-01 (1/4)
          eep: M5-38-08
          sender: 
            id: FF-88-03-01
            eep: A5-38-08

        cover:
        - id: FF-8F-AB-96
          name: FSB14 FF-8F-AB-96 (1/2)
          eep: G5-3F-7F
          sender: 
            id: FF-88-03-40
            eep: H5-3F-7F
          device_class: shutter
          time_closes: 25
          time_opens: 25
```

Sender ID is in the PCT14 with function 31.

I tried to understand the Error Message, but couldnt figure if the problem is based on my config or some other problem with the integration.

![97795e49bb280d7187afa36bcd357c2a9cbce2f7](https://github.com/grimmpp/home-assistant-eltako/assets/12120359/f4b00518-9c64-41bb-bb74-231f0a9bd180)

**Comment by schlegel** [comment]
I have exactly the same error "Received ESP3 response with return code WRONG_PARAM (3) - 0x02 ['0x3'] [] OrderedDict()". 

Can I help somehow with more information?

**Comment by grimmpp** [comment]
Can you let me know when it happens and which gateway and version of the integration are you using?

**Comment by schlegel** [comment]
I'm using a usb300 gateway and the feature-branch version of the integration. It happens when I click a button (up or down) on a cover.

**Comment by schlegel** [comment]
<img width="333" alt="Screenshot 2024-07-11 at 23 43 10" src="https://github.com/user-attachments/assets/7176df71-cb63-4b1f-af35-f72d2c8b53a4">
<img width="674" alt="Screenshot 2024-07-11 at 23 43 16" src="https://github.com/user-attachments/assets/6d5834aa-66e1-4933-9e85-ff05fb1e9a42">

**Comment by schlegel** [comment]
I kinda found the problem. I set the base id to FF-80-80-00 (like shown in https://github.com/grimmpp/home-assistant-eltako/tree/main/docs/gateways). After some crazy debug with logging, I found out that I have a total different base id. Setting the correct base id fixed everything. Do you know a better way how to get the base id of the usb stick? Might be a good addition to the documentation. 

<img width="896" alt="Screenshot 2024-07-13 at 01 02 48" src="https://github.com/user-attachments/assets/396e9a57-b3ac-49d1-a31d-fbfae826eda3">

**Comment by grimmpp** [comment]
There are commands and tools to query the base_id via serial protocol.

The easiest way I know is to use eo_man. Just plugin [ESP300](https://github.com/grimmpp/enocean-device-manager/tree/main) and it should work similar to FAM14 like described in [here](https://github.com/grimmpp/enocean-device-manager/tree/main/docs/getting-started).

**Comment by grimmpp** [comment]
I've looked into the code but I don't see a relation between `base_id `and the cover command. Did you also use a `sender id` which was not based on the `base_id`?

**Comment by schlegel** [comment]
yes, both base_id and therefore also sender_id were not aligned with my esp300. as the bus can query the base_id of the stick, maybe you know a way to show it in home assistant. could help some people

**Comment by grimmpp** [comment]
Theoretically there would be a way but only for a new gateways. And the
question is when to display. Actually you need to provide the config
beforehand.
That's why I developed eo_man which can autogenerated a default config
containing all the things configured in your bus devices.

Kai Schlegel ***@***.***> schrieb am Mi., 17. Juli 2024,
21:59:

**Comment by schlegel** [comment]
maybe a warning at startup if the base ids are not matching? anyways, it was not a bug, but only a misuse from my side. i hope it‘s the same for the original author of this bug

**Comment by grimmpp** [comment]
I've tried that earlier. At this time the com-lib did not provided it. 
I've put in onto my backlog.

**Comment by grimmpp** [comment]
This report targets Eltako v1.4.2 and the old USB300/ESP3 conversion path. It is superseded by the current ESP3 gateway investigation in #185 and the current MGW-LAN/ESP2 issue in #204. Closing this obsolete duplicate for versions after 2.2.
